sphinx/transforms/post_transforms/__init__.py

```diff
@@ -95,6 +95,7 @@ def run(self, **kwargs: Any) -> None:
                     newnode = self.resolve_anyref(refdoc, node, contnode)
                 # no new node found? try the missing-reference event
                 if newnode is None:
+                    node['refdoc'] = refdoc
```

Comment:
I feel it's better to fill this field before calling `domain.resolve_xref()`. So I prefer to call `node.setdefault('refdoc', self.env.docname)` at line 81. What you think?

Reply:
That is also totally fine by me.